transport/src/main/java/io/netty/channel/SingleThreadEventLoop.java

```diff
@@ -182,5 +163,5 @@ public int registeredChannels() {
     /**
      * Marker interface for {@link Runnable} that will not trigger an {@link #wakeup(boolean)} in all cases.
      */
-    interface NonWakeupRunnable extends Runnable { }
+    interface NonWakeupRunnable extends SingleThreadEventExecutor.NonWakeupRunnable { }
```

Comment:
remove this interface all together and just use `SingleThreadEventExecutor.NonWakeupRunnable` directly in other classes ?

Reply:
ah never mind this will not work as it is protected..

Reply:
Right exactly... alternative would be to just change `SingleThreadEventExecutor.NonWakeupRunnable` to be public, I don't think there would be much harm in doing that, wdyt?

Reply:
lets keep it protected for now